tasks/console-errors.js

```diff
@@ -0,0 +1,103 @@
+const puppeteer = require('puppeteer'); // eslint-disable-line
+const { getSitemapUrls } = require('./get-sitemap-links'); // eslint-disable-line
+
+// Here we are excluding shortbread errors because these are domain specific and are expected to fail in a local environment
+const excludedErrors = [
+  {
+    type: 'Shortbread',
+    errorText:
+      "Shortbread failed to set user's cookie preference because the domain name that was passed"
+  }
+];
+
+const excludedScripts = [
+  'prod.assets.shortbread.aws',
+  'prod.tools.shortbread.aws',
+  'prod.tools.shortbread.aws.dev',
+  'aa0.awsstatic.com',
+  'alpha.d2c.marketing.aws.dev',
+  'vs-alpha.aws.amazon.com'
+];
+
+const LOCALHOST = 'http://localhost:3000';
+
+const checkPage = async (url) => {
+  const errorsFound = [];
+  const browser = await puppeteer.launch({ headless: 'new' });
+
+  const page = await browser.newPage();
+
+  await page.setRequestInterception(true);
+
+  page
+    .on('pageerror', (message) => {
+      const errorText = message.message;
+      const excluded = excludedErrors.some((excludedError) => {
+        return errorText.includes(excludedError.errorText);
+      });
+
+      if (!excluded) {
+        errorsFound.push({
+          page: url,
+          message: errorText
+        });
+      }
+    })
+    .on('console', (message) => {
+      if (message.type().toLowerCase() === 'error') {
+        const errorText = message.text();
+        const callingScript = message.location().url;
+        const excludedFromError = excludedErrors.some((excludedError) => {
+          return errorText.includes(excludedError.errorText);
+        });
+        const excludedFromScript = excludedScripts.some((excludedScript) => {
+          return callingScript.includes(excludedScript);
+        });
+        const excluded = excludedFromError || excludedFromScript;
+
+        if (!excluded) {
+          errorsFound.push({
+            page: url,
+            message: errorText,
+            stackTrace: message.stackTrace()
+          });
+        }
+      }
+    })
+
+    .on('request', (interceptedRequest) => {
+      const excludedFromScript = excludedScripts.some((excludedScript) => {
+        return interceptedRequest.url().includes(excludedScript);
+      });
+      if (excludedFromScript) {
+        interceptedRequest.abort();
+      } else interceptedRequest.continue();
+    });
```

Comment:
Why do we need to handle aborting the requests when we already check for it in the `console` event for the `excludedScripts`?

Reply:
There were errors that were making it past the console check even though they were originating in the excluded scripts.  Aborting the requests for the scripts has proven to eliminate those errors and should make the console error check for these scripts redundant.  The console check will be removed, thanks.

Reply:
@jacoblogan and I found that ultimately, the console check is necessary to filter out the 'failed to load script' error we hit without it.